server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -83,9 +83,24 @@ public class RemoteClusterStateService implements Closeable {
 
     private static final Logger logger = LogManager.getLogger(RemoteClusterStateService.class);
 
-    // TODO make this two variable as dynamic setting [issue: #10688]
-    public static final int INDEX_METADATA_UPLOAD_WAIT_MILLIS = 20000;
-    public static final int GLOBAL_METADATA_UPLOAD_WAIT_MILLIS = 20000;
+    // default value for index metadata upload wait is 20s
+    static volatile TimeValue INDEX_METADATA_UPLOAD_TIMEOUT = TimeValue.timeValueMillis(20000);
+    // default value for global metadata upload wait is 20s
+    static volatile TimeValue GLOBAL_METADATA_UPLOAD_TIMEOUT = TimeValue.timeValueMillis(20000);
+
+    public static final Setting<TimeValue> INDEX_METADATA_UPLOAD_TIMEOUT_SETTING = Setting.timeSetting(
+        "cluster.remote_store.index_metadata.upload_timeout",
```

Comment:
let's add `state` as well in name like `cluster.remote_store.state.index_metadata.upload_timeout` same for global metadata. That way we can make all remote cluster state related settings consistent.

Reply:
makes sense, thanks for the callout, I will make this change